packages/firestore/src/api/database.ts

```diff
@@ -703,10 +705,31 @@ export class Firestore implements PublicFirestore, FirebaseService {
 
   INTERNAL = {
     delete: async (): Promise<void> => {
-      // The client must be initalized to ensure that all subsequent API usage
+      // The client must be initialized to ensure that all subsequent API usage
       // throws an exception.
       this.ensureClientConfigured();
-      await this._firestoreClient!.terminate();
+
+      this._queue.enterRestrictedMode();
```

Comment:
This and similar changes pulling FirestoreClient code into the API layer seem like a fairly radical departure from the way the other SDKs work.

Notably, on other platforms we have an API layer that serves as a stable interface to user code and we have most of the functionality in places like FirestoreClient. I would expect that even after we transition to tree-shakeable form, we'd still have much of the same structure, except that instead of a FirestoreClient class with methods, it would be a FirestoreClient struct with free functions. Moving this code out of FirestoreClient into the classic API layer seems like the wrong move--this code should end up somewhere in the exp code in the FirestoreClient equivalent over there. If we don't do that, isn't this going to have to be duplicated there?

Reply:
I addressed this comment in a new PR and will abandon this one for now: https://github.com/firebase/firebase-js-sdk/pull/3974